pkg/controllers/subnet/subnet_poll.go

```diff
@@ -0,0 +1,254 @@
+
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/vpc/v1alpha1"
+)
+
+var ticker = time.NewTicker(10 * time.Minute)
+
+// pollSharedSubnets periodically polls NSX for shared subnet status updates
+// It runs in a separate goroutine and polls every 10 minutes.
+// The polling can be stopped by sending a value to the stopCh channel.
+func (r *SubnetReconciler) pollSharedSubnets(stopCh chan bool) {
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			r.pollAllSharedSubnets()
+		case <-stopCh:
+			log.Info("Stopping shared subnet polling")
+			return
+		}
+	}
+}
+
+// pollAllSharedSubnets polls NSX for all shared subnets in the queue.
+// It groups subnets by associatedResource to avoid redundant NSX API calls.
+// For each unique associatedResource, it gets the NSX subnet and status only once,
+// then updates all related Subnet CRs with the same information.
+// It also removes subnets from the queue if they no longer exist or are being deleted.
+func (r *SubnetReconciler) pollAllSharedSubnets() {
+	// Group namespacedNames by associatedResource
+	resourceMap := make(map[string][]types.NamespacedName)
+
+	// Create a read lock to safely iterate through the map
+	r.sharedSubnetsMutex.RLock()
+	for namespacedName, associatedResource := range r.sharedSubnetsMap {
+		resourceMap[associatedResource] = append(resourceMap[associatedResource], namespacedName)
+	}
+	r.sharedSubnetsMutex.RUnlock()
+
+	// Process each unique associatedResource
+	for associatedResource, namespacedNames := range resourceMap {
+		ctx := context.Background()
+		log.Info("Polling shared subnets", "AssociatedResource", associatedResource, "SubnetCount", len(namespacedNames))
+
+		// Get the first subnet's namespace to use for the NSX API call
+		// We need to get the actual subnet to determine its namespace
+		var namespace string
+		var validSubnets []types.NamespacedName
+
+		for _, namespacedName := range namespacedNames {
+			subnetCR := &v1alpha1.Subnet{}
+			if err := r.Client.Get(ctx, namespacedName, subnetCR); err != nil {
+				r.handleSubnetGetError(err, namespacedName)
+				continue
+			}
+
+			// Skip if the subnet is being deleted
+			if !subnetCR.DeletionTimestamp.IsZero() {
+				r.removeSubnetFromPollingQueue(namespacedName, "deleting")
+				continue
+			}
+
+			namespace = subnetCR.Namespace
+			validSubnets = append(validSubnets, namespacedName)
+		}
+
+		if namespace == "" || len(validSubnets) == 0 {
+			log.Info("No valid subnets found for associated resource", "AssociatedResource", associatedResource)
+			continue
+		}
+
+		// Get the NSX subnet based on the associated resource - only once per associatedResource
+		nsxSubnet, err := r.SubnetService.GetNSXSubnetByAssociatedResource(associatedResource)
```

Comment:
in subnet Poll goroutine, for each shared subnet, after we find all the Subnet CRs using that shared subnet, 
can we just enqueue the Subnet CR  and let Subnet Controller to handle it?
We do not need to update Subnet here.

Reply:
fixed